autorest/codegen/__init__.py

```diff
@@ -111,12 +116,34 @@ def _build_exceptions_set(yaml_data: List[Dict[str, Any]]) -> Set[int]:
                     exceptions_set.add(id(exception["schema"]))
         return exceptions_set
 
+    @staticmethod
+    def _build_security(yaml_data: Dict[str, Any]) -> Dict[str, Any]:
+        security_yaml = yaml_data.get("security")
+        security: Dict[str, Any] = dict()
+        if security_yaml and security_yaml.get("authenticationRequired"):
+            # the key word is "AADToken" and "AzureKey"
+            aad_token = set()
+            azure_key = ""
+            for scheme in security_yaml.get("schemes"):
+                if AAD_TOKEN == scheme["type"]:
+                    aad_token.update(scheme["scopes"])
+                elif AZURE_KEY == scheme["type"]:
+                    # only accept the last one
+                    azure_key = scheme["headerName"]
+            if aad_token:
+                security[AAD_TOKEN] = list(aad_token)
+            if azure_key:
+                security[AZURE_KEY] = azure_key
+        return security
+
     def _create_code_model(self, yaml_data: Dict[str, Any], options: Dict[str, Union[str, bool]]) -> CodeModel:
         # Create a code model
 
-        code_model = CodeModel(options=options)
+        code_model = CodeModel(options=options, security=self._build_security(yaml_data))
         if code_model.options['credential']:
             self._handle_default_authentication_policy(code_model)
+        elif code_model.security:
```

Comment:
I think it would be better to set a couple of credential properties on code model that represent the credential and credential policy completely. We don't want to add a whole new proprty for security, instead we want to update code models' credential and credential policy information

So we start off by setting these properties using information from the security flag
Then, if users pass in credential flags, we set the corresponding properties with the credential flag information
Finally, if users passed in azure-arm, we update the properties with what we want for azure-arm. 

That way, the end result of code model is the same amount of properties, and we have the correct priority between these ways of specifying security
 first with security, then with credential flags (if users specified them), and finally set azure-arm. That way each properly overrides the other

Reply:
@msyyc it might be better to first do a PR supporting two kinds of authentication for the current flags, then doing this PR, because we should treat these as generating the same code, just different ways of inputting.

Reply:
It is ok to do a PR before this PR to support two kinds of auth. I think it is also fine to merge this PR (Java has supported it long time ago) and then refactor the code if needed. I think it is better to support security before GA so I prefer the latter. cc @Imazuel